hbase-client/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKAssign.java

```diff
@@ -868,7 +868,15 @@ public static int transitionNode(ZooKeeperWatcher zkw, HRegionInfo region,
     try {
       rt = RegionTransition.createRegionTransition(
           endState, region.getRegionName(), serverName, payload);
-      if(!ZKUtil.setData(zkw, node, rt.toByteArray(), stat.getVersion())) {
+      boolean isDataSet;
+      try {
+        isDataSet = ZKUtil.setData(zkw, node, rt.toByteArray(), stat.getVersion());
+      } catch (KeeperException.BadVersionException e) {
+        isDataSet = false;
+        LOG.info("Received BadVersionException from ZK for " + encoded
```

Comment:
nit: Always better to use parameterized logging for performance [1]. Here info is the default, so probably doesn't matter as much. 

[1] https://logging.apache.org/log4j/2.x/performance.html

Reply:
That is true but for most of the cases of all branch-1s use `org.apache.commons.logging.Log`

Reply:
I see ok. Looks like branch-1.3 also pulls in slf4j dependency and quite a few places use it. For ex: https://github.com/apache/hbase/blob/branch-1.3/hbase-server/src/main/java/org/apache/hadoop/hbase/tool/CanaryTool.java#L428 but I see your point that there is no standard usage pattern there. Anyway this comment was not meant to be a blocker for merging. Feel free to fix it if you have free cycles.

Reply:
I agree we can move to slf4j logger and improve this, but may be in another Jira :)